tests/test_transforms/test_decomposition/test_change_points_segmentation_transform.py

```diff
@@ -0,0 +1,160 @@
+
+
+def test_transform(pre_transformed_df: pd.DataFrame):
+    """Check that transform method generate new column."""
+    out_column = "result"
+    bs = _OneSegmentChangePointsSegmentationTransform(
+        in_column="target", change_point_model=Binseg(), out_column=out_column, n_bkps=5
+    )
+    bs.fit(df=pre_transformed_df["segment_1"])
+    transformed = bs.transform(df=pre_transformed_df["segment_1"])
+    assert set(transformed.columns) == {"target", out_column}
+    assert transformed[out_column].dtype == "category"
+
+
+def test_inverse_transform(pre_transformed_df: pd.DataFrame):
+    """Check that inverse_transform works."""
+    out_column = "result"
+    bs = _OneSegmentChangePointsSegmentationTransform(
+        in_column="target", change_point_model=Binseg(), out_column=out_column, n_bkps=5
+    )
+    bs.fit(df=pre_transformed_df["segment_1"])
+
+    transformed = bs.transform(df=pre_transformed_df["segment_1"].copy(deep=True))
+    bs.inverse_transform(transformed)
+
+
+def test_monotonously_result(pre_transformed_df: pd.DataFrame):
+    """Check that resulting column is monotonously non-decreasing."""
+    out_column = "result"
+    bs = _OneSegmentChangePointsSegmentationTransform(
+        in_column="target", change_point_model=Binseg(), out_column=out_column, n_bkps=5
+    )
+    bs.fit(df=pre_transformed_df["segment_1"])
+
+    transformed = bs.transform(df=pre_transformed_df["segment_1"].copy(deep=True))
+    result = transformed[out_column].astype(int).values
+    assert (result[1:] - result[:-1] >= 0).mean() == 1
+
+
+def test_transform_raise_error_if_not_fitted(pre_transformed_df: pd.DataFrame):
+    """Test that transform for one segment raise error when calling transform without being fit."""
+    out_column = "result"
+    transform = _OneSegmentChangePointsSegmentationTransform(
+        in_column="target", change_point_model=Binseg(), out_column=out_column, n_bkps=5
+    )
+    with pytest.raises(ValueError, match="Transform is not fitted!"):
+        _ = transform.transform(df=pre_transformed_df["segment_1"])
+
+
+def test_fit_transform_with_nans_in_middle_raise_error(df_with_nans):
+    out_column = "result"
+    bs = _OneSegmentChangePointsSegmentationTransform(
+        in_column="target", change_point_model=Binseg(), out_column=out_column, n_bkps=5
+    )
+    with pytest.raises(ValueError, match="The input column contains NaNs in the middle of the series!"):
+        _ = bs.fit_transform(df=df_with_nans["segment_1"])
+
+
+def test_backtest(simple_ar_ts):
+    model = CatBoostModelPerSegment()
+    horizon = 3
+    out_column = "result"
+    bs = ChangePointsSegmentationTransform(
+        in_column="target", change_point_model=Binseg(), out_column=out_column, n_bkps=5
+    )
+    pipeline = Pipeline(model=model, transforms=[bs], horizon=horizon)
+    _, _, _ = pipeline.backtest(ts=simple_ar_ts, metrics=[SMAPE()], n_folds=3)
+
+
+def test_future_and_past_filling(simple_ar_ts):
```

Comment:
I guess here is more sensible to do the make_future and check that transform create feature in the future

Reply:
Add the separate test for make_future